src/Connections/GithubRepo.ts

```diff
@@ -1042,6 +1052,32 @@ ${event.release.body}`;
         }
     }
 
+    public static async getConnectionTargets(userId: string, tokenStore: UserTokenStore, config: BridgeConfigGitHub): Promise<GitHubRepoConnectionTarget[]> {
+        // Search for all repos under the user's control.
+        const octokit = await tokenStore.getOctokitForUser(userId);
+        if (!octokit) {
+            throw new ApiError("User is not authenticated with GitHub", ErrCode.ForbiddenUser);
+        }
+        const allRepos = await octokit.repos.listForAuthenticatedUser({
```

Comment:
Does this only return repos the user can actually control (i.e. create webhooks for), or all the repos the user can see?

Reply:
The docstring says it's the latter:
> Lists repositories that the authenticated user has explicit permission (:read, :write, or :admin) to access.
> 
> The authenticated user has explicit permission to access repositories they own, repositories where they are a collaborator, and repositories that they can access through an organization membership.

Is write permission needed for adding read-only webhooks?

Reply:
Surely you need at least write permissions, as `read` doesn't grant you the ability to edit the settings?